go/ioutil2/ioutil.go

```diff
@@ -20,11 +20,16 @@ func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
 		return err
 	}
 	n, err := f.Write(data)
+	if err == nil {
```

Comment:
I actually don't think it's needed because Close is supposed to sync. But it's harmless.